compiler/rustc_mir_build/src/check_unsafety.rs

```diff
@@ -339,8 +339,16 @@ impl<'a, 'tcx> Visitor<'a, 'tcx> for UnsafetyVisitor<'a, 'tcx> {
         };
 
         match &pat.kind {
-            PatKind::Leaf { .. } => {
+            PatKind::Leaf { subpatterns, .. } => {
                 if let ty::Adt(adt_def, ..) = pat.ty.kind() {
+                    for pat in subpatterns {
+                        let field = &pat.field;
+                        if adt_def.variants().iter().next().unwrap().fields[*field].safety
```

Comment:
I think you should use `non_enum_variant` here:

```suggestion
                        if adt_def.non_enum_variant().fields[*field].safety
```

I don't think(?) we use `PatKind::Leaf` for enums ever.

Reply:
You're correct - I could not find the correct method ;)